### PR TITLE
add week2 hw #Class22 Salih

### PR DESCRIPTION
**Pull request by salih18**

### CODE REVIEW

**Comment by yash-kapila** [comment]
Hi @salih18 - I don't see homework 1.1 (doubling odd numbers) submitted. Could you please check?

**Comment by salih18** [comment]
Sorry @yash-kapila I have just noticed I skipped doing it. I have started from 1.2. i am gonna do it now.

**Comment by salih18** [comment]
@yash-kapila I have just done it, and made PR.

File: Week2/homework/map-filter.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
 function doubleOddNumbers(numbers) {
-  // Replace this comment and the next line with your code
-  console.log(numbers);
+  const arr = numbers.filter(num => num % 2 !== 0).map(num => num * 2);
+  return arr;
```
Comment (yash-kapila) [inline Week2/homework/map-filter.js]: This is only my personal opinion but you can also combine these two statements and simply have
```
return numbers.filter(num => num % 2 !== 0).map(num => num * 2);
```

File: Week2/homework/maartjes-work.js [hunk 2]
```diff
@@ -46,16 +46,20 @@ const maartjesTasks = monday.concat(tuesday);
...
 
 function computeEarnings(tasks, hourlyRate) {
-  // Replace this comment and the next line with your code
-  console.log(tasks, hourlyRate);
+  const calc = tasks
+    .map(task => ({ name: task.name, duration: task.duration / 60 })) // transforms minutes to hours
+    .filter(time => time.duration >= 2) // filter array task duration equal and greater than 2
+    .reduce((acc, curr) => acc + curr.duration * hourlyRate, 0); // add duration * hourly rate values to the acc
+
+  return calc;
```
Comment (yash-kapila) [inline Week2/homework/maartjes-work.js]: Quite nice - clean and concise. I like it.

However, I am not too happy about the variable names `calc` and `curr`. I think `totalEarnings` and `task` respectively might better describe your intentions. 😉

**Comment by salih18** [comment]
@yash-kapila thank you for your review. I have just updated my codes as a feedback.

**Pull request closed without merging**